mno_analysis_tools/window_of_downtime.py

```diff
@@ -0,0 +1,70 @@
+# Compute maximum window of time with 0 messages
+import json
+import pytz
+import iso8601
+from datetime import datetime
+from temba_client.v2 import Message
+from core_data_modules.logging import Logger
+
+
+def group(lst, n):
+    """group([0,3,4,10,2,3], 2) => [(0,3), (4,10), (2,3)]
+    
+    Group a list into consecutive n-tuples. Incomplete tuples are
+    discarded e.g.
+    
+    >>> group(range(10), 3)
+    [(0, 1, 2), (3, 4, 5), (6, 7, 8)]
+    """
+    return zip(*[lst[i::n] for i in range(n)])
+
+
+log = Logger(__name__)
+log.set_project_name("WindowOfDowntime")
+
+if __name__ == "__main__":
+    with open("data.txt", mode="r") as f:
+        file_content = f.readline()
+        output = json.loads(file_content)
+        f.close()
+
+    data = [Message.deserialize(val) for val in output]
```

Comment:
```suggestion
    messages = [Message.deserialize(val) for val in output]
```

The use of `data` in the pipelines was a mistake we're still paying for in the pipelines because we weren't sure what the data would contain when we started. If you know these are messages, call them messages.

Reply:
Noted..